src/base/utils/fs.h

```diff
@@ -33,12 +33,40 @@
  * Utility functions related to file system.
  */
 
+#include <iterator>
+#include <memory>
+
 #include <QString>
 
+class QByteArray;
+class QIODevice;
+
 namespace Utils
 {
     namespace Fs
     {
+        // A wrapper class that satisfy LegacyOutputIterator requirement
+        class FileOutputIterator
+            : public std::iterator<std::output_iterator_tag, void, void, void, void>
+        {
+        public:
+            explicit FileOutputIterator(QIODevice &device, const int bufferSize = (4 * 1024));
+            FileOutputIterator(const FileOutputIterator &other) = default;
+            ~FileOutputIterator();
+
+            // mimic std::ostream_iterator behavior
+            FileOutputIterator &operator=(const char c);
+            // TODO: make these `constexpr` in C++17
+            FileOutputIterator &operator*();
+            FileOutputIterator &operator++();
+            FileOutputIterator &operator++(int);
+
+        private:
+            QIODevice *m_device;
+            std::shared_ptr<QByteArray> m_buffer;
```

Comment:
Isn't it overkill to use shared_ptr here?

Reply:
I don't like it either, however it is required to satisfy CopyConstructible requirement in LegacyOutputIterator.